### ISSUE TITLE
pre define shunt backend in case routegroups are used

### ISSUE THREAD

**Issue by szuecs** [zalando-incubator/stackset-controller#289]
I am not sure if it makes sense, but a user had the idea to pre define shunt backend in case routegroups are used, such that they don't need to define additionalBackends. I documented it now in https://github.com/zalando-incubator/stackset-controller/pull/288

**Comment by mikkeloscar** [comment]
I had the same idea when implementing RouteGroup support and agree that this could reduce the boiler plate needed by users. What I wonder though if is `shunt` could be made special in RouteGroup in general so you don't have to define it also in a plain RouteGroup (outside of StackSet)?

**Comment by szuecs** [comment]
@mikkeloscar I don't think so, because for RouteGroup in skipper this is just one of our supported backends and why should we add special things for `shunt` and not for `loopback`. I think the less magic a building block is the better.
Adding this to stackset makes sense because we abstract here ingress/routegroups anyways.

**Comment by mikkeloscar** [comment]
From my perspective any of the backend types where there is no additional configuration possible could be automatically available.

> I think the less magic a building block is the better.

I agree when we think of it as a building block. However when users has a desire to write RouteGroups directly, then I can see the value in some "magic" such that it becomes less verbose.

> Adding this to stackset makes sense because we abstract here ingress/routegroups anyways.

Agreed
